src/Ast/Ast.dfy

```diff
@@ -490,6 +497,36 @@ module Ast {
           DeclsToString(vv) +
           Pattern.ListToString(patterns) + " " + body.ToString(opStrength.SubexpressionPower(PrintUtil.Right, context))
         )
+      case ClosureExpr(closureBindings, resultVar, resultType, properties) =>
+        "lift " + ClosureBindingsToString(closureBindings) + " into " + resultVar + ": " + resultType.ToString() + " by { " + ClosurePropertiesToString(properties) + " }"
```

Comment:
This will leave two spaces between `lift` and `into` if `closureBindings` is empty. Instead, you could change `"lift "` to `"lift"` and arrange for `ClosureBindingsToString` to prefix every binding with a space.